arrow-core/src/main/kotlin/arrow/core/extensions/option.kt

```diff
@@ -82,24 +78,20 @@ interface OptionSemigroup<A> : Semigroup<Option<A>> {
     }
 }
 
-@extension
 interface OptionSemigroupal : Semigroupal<ForOption> {
   override fun <A, B> Kind<ForOption, A>.product(fb: Kind<ForOption, B>): Kind<ForOption, Tuple2<A, B>> =
     fb.fix().ap(this.map { a: A -> { b: B -> Tuple2(a, b) } })
 }
 
-@extension
 interface OptionMonoidal : Monoidal<ForOption>, OptionSemigroupal {
   override fun <A> identity(): Kind<ForOption, A> = None
 }
 
-@extension
 interface OptionMonoid<A> : Monoid<Option<A>>, OptionSemigroup<A> {
   override fun SG(): Semigroup<A>
   override fun empty(): Option<A> = None
 }
 
-@extension
 interface OptionApplicativeError : ApplicativeError<ForOption, Unit>, OptionApplicative {
```

Comment:
```
@Deprecated(
  message = "ApplicativeError typeclass is deprecated and will be removed in 0.13.0. Use concrete methods on Const",
  level = DeprecationLevel.WARNING
)
```

We should add this for all the deprecated interface instances. We need to do this retroactively for the already done PRs, I think we forgot to do it there too.